google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java

```diff
@@ -0,0 +1,155 @@
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineStandardRunTask;
+import com.google.cloud.tools.intellij.appengine.server.instance.AppEngineServerModel;
+import com.google.cloud.tools.intellij.util.GctBundle;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.process.OSProcessHandler;
+import com.intellij.execution.runners.ProgramRunner;
+import com.intellij.javaee.run.configuration.CommonModel;
+import com.intellij.javaee.run.localRun.EnvironmentHelper;
+import com.intellij.javaee.run.localRun.ExecutableObject;
+import com.intellij.javaee.run.localRun.ExecutableObjectStartupPolicy;
+import com.intellij.javaee.run.localRun.ScriptHelper;
+import com.intellij.javaee.run.localRun.ScriptsHelper;
+import com.intellij.packaging.artifacts.Artifact;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * Runs a Google App Engine Standard app locally with devappserver, through the tools lib.
+ */
+public class CloudSdkStartupPolicy implements ExecutableObjectStartupPolicy {
+
+  // The startup process handler is kept so the process can be explicitly terminated, since we're
+  // not delegating that to the framework.
+  OSProcessHandler startupProcessHandler;
+
+  @Nullable
+  @Override
+  public ScriptHelper createStartupScriptHelper(ProgramRunner programRunner) {
+    return new ScriptHelper() {
+      @Nullable
+      @Override
+      public ExecutableObject getDefaultScript(final CommonModel commonModel) {
+        return new ExecutableObject() {
+          @Override
+          public String getDisplayString() {
+            return GctBundle.getString("appengine.run.startupscript.name");
+          }
+
+          @Override
+          public OSProcessHandler createProcessHandler(String string, Map<String, String> map)
+              throws ExecutionException {
+            AppEngineServerModel runConfiguration;
+
+            try {
+              // Getting the clone so the debug flags aren't added to the persisted settings.
+              runConfiguration = (AppEngineServerModel) commonModel.getServerModel().clone();
+            } catch (CloneNotSupportedException ee) {
+              throw new ExecutionException(ee);
+            }
+
+            // This is the place we have access to the debug jvm flags provided by IJ in the
+            // Startup/Shutdown tab. We need to add it here.
+            String jvmDebugFlag = map.get("");
+            if (jvmDebugFlag != null) {
+              runConfiguration.addAllJvmFlags(Arrays.asList(jvmDebugFlag.trim().split(" ")));
+            }
+
+            AppEngineStandardRunTask runTask =
+                new AppEngineStandardRunTask(runConfiguration);
+            AppEngineExecutor executor = new AppEngineExecutor(runTask);
+            executor.run();
+
+            Process devappserverProcess = executor.getProcess();
```

Comment:
Got rid of this variable since we're now checking SDK validity at the beginning of this method, not AppEngineStandardRunTask.